compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -126,24 +131,45 @@ def start_options(f):
     return f
 
 
+def get_endpoint_by_name_or_uuid(ctx, param, value):
+    path = Endpoint.get_endpoint_by_name_or_uuid(get_config_dir(), value)
+    if path is None:
+        try:
+            uuid.UUID(value)
+        except ValueError:
+            msg = NOT_CONFIGURED_ERR_MSG.format(ep_name=value)
+        else:
+            msg = NO_EP_WITH_ID_ERR_MSG.format(ep_id=value)
+
+        raise ClickException(msg)
+    ctx.params["ep_dir"] = path
```

Comment:
I think this could use some refactoring to consolidate and simplify things. Below is my suggestion:
```python
def get_endpoint_by_name_or_uuid(ctx, param, value):
    conf_dir = get_config_dir()
    try:
        uuid.UUID(value)
    except ValueError:
        # We are dealing with an EP name
        path = conf_dir / value
    else:
        path = Endpoint.get_endpoint_dir_by_uuid(conf_dir, value)
        if path is None:
            msg = textwrap.dedent(
                f"""
                There is no endpoint on this machine with ID '{value}'!

                1. Please create a configuration template with:
                    globus-compute-endpoint configure
                2. Update the configuration
                3. Start the endpoint
                """
            )
            raise ClickException(msg)
    ctx.params["ep_dir"] = path
```

```python
# globus_compute_endpoint/endpoint/endpoint.py

class Endpoint:
    # ...
    @staticmethod
    def get_endpoint_dir_by_uuid(
        gc_conf_dir: pathlib.Path, uuid: uuid.UUID | str
    ) -> pathlib.Path | None:
        for ep_path in Endpoint._get_ep_dirs(gc_conf_dir):
            if uuid == Endpoint.get_endpoint_id(ep_path):
                return ep_path
        return None
    # ...
```